src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        if (fruitTransaction.getFruit() == null) {
+            throw new RuntimeException("There are no "
+                    + fruitTransaction.getFruit() + " int the storage");
+        }
+        int amountInStore = FruitStorage.getFruits().get(fruitTransaction.getFruit());
+        int amountToBuy = fruitTransaction.getQuantity();
+        if (amountInStore < amountToBuy) {

```

Comment:
The balance check is correct, but you should also handle the case where the fruit is not present in the storage at all, which is currently not covered by the code.